test/libraries/Pool.t.sol

```diff
@@ -41,48 +41,159 @@ contract PoolTest is Test, GasSnapshot {
         }
     }
 
-    function testModifyLiquidity(
+    function test_initialize_updatesState() public {
+        uint160 sqrtPriceX96 = Constants.SQRT_PRICE_1_1;
+        uint24 protocolFee = MAX_PROTOCOL_FEE;
+        uint24 lpFee = 3000;
+        state.initialize(sqrtPriceX96, protocolFee, lpFee);
+        assertEq(state.slot0.sqrtPriceX96(), sqrtPriceX96);
+        assertEq(state.slot0.protocolFee(), protocolFee);
+        assertEq(state.slot0.tick(), TickMath.getTickAtSqrtPrice(sqrtPriceX96));
+        assertLt(state.slot0.tick(), TickMath.MAX_TICK);
+        assertGt(state.slot0.tick(), TickMath.MIN_TICK - 1);
+    }
+
+    function test_initialize_revertsWithInvalidSqrtPrice_tooLow() public {
+        uint160 sqrtPriceX96 = TickMath.MIN_SQRT_PRICE - 1;
```

Comment:
would be great to do complete bounds checking!
so `MIN_SQRT_PRICE` doesnt revert etc

Reply:
and that `MAX_SQRT_PRICE-1` succeeds